spec/hydra/pcdm/models/object_spec.rb

```diff
@@ -2,28 +2,228 @@
 
 describe Hydra::PCDM::Object do
 
-  let(:object1) { Hydra::PCDM::Object.new }
-  let(:object2) { Hydra::PCDM::Object.new }
-  let(:object3) { Hydra::PCDM::Object.new }
+  describe '#objects=, +=' do
+    context 'with acceptable child objects' do
+      let(:object1) { described_class.new }
+      let(:object2) { described_class.new }
+      let(:object3) { described_class.new }
+      let(:object4) { described_class.new }
+      let(:object5) { described_class.new }
 
-  describe '#objects=' do
-    it 'should aggregate objects' do
-      object1.objects = [object2, object3]
-      object1.save
-      expect(object1.objects).to eq [object2, object3]
+      it 'should add objects objects' do
+        subject.child_objects = [object1,object2]
+        subject.child_objects += [object3]
+        subject.child_objects += [object4,object5]
+        expect( subject.child_objects ).to eq [object1,object2,object3,object4,object5]
+     end
+
+      it 'should allow sub-objects' do
+        subject.child_objects = [object1,object2]
+        object1.child_objects = [object3]
+        expect( subject.child_objects ).to eq [object1,object2]
+        expect( object1.child_objects ).to eq [object3]
+      end
+
+      it 'should allow repeating objects' do
+      skip( "pending resolution of PCDM-105 and AF-853") do
+        subject.child_objects = [object1,object2]
+        subject.child_objects += [object1]
+        expect( subject.child_objects ).to eq [object1,object2,object1]
+      end
+      end
+
+      describe 'adding objects that are ancestors' do
```

Comment:
It seems to me that all four of the tests in this block are hitting the same code path. Is that right? Do we need all of that? Is there some reason we can't rely on the unit tests? https://github.com/projecthydra-labs/hydra-pcdm/blob/e0bcaea6e2fd2c7c1d09a9495d58e8fd2fcccb55/spec/hydra/pcdm/ancestor_checker_spec.rb